api/src/database/migrations/20231009A-update-csv-fields-to-text.ts

```diff
@@ -0,0 +1,47 @@
+import type { Knex } from 'knex';
+import { getHelpers } from '../helpers/index.js';
+import { createInspector } from '@directus/schema';
+
+export async function up(knex: Knex) {
+	const inspector = createInspector(knex);
+	const helper = getHelpers(knex).schema;
+	const csvFields = await knex.select('collection', 'field').from('directus_fields').where('special', '=', 'cast-csv');
+
+	const updates: Promise<void>[] = [];
+
+	for (const { collection, field } of csvFields) {
+		updates.push(
+			inspector.columnInfo(collection, field).then((column) => {
+				if (column.data_type === 'text') return;
+
+				return helper.changeToType(collection, field, 'text', {
+					default: column.default_value,
+					nullable: column.is_nullable,
+				});
+			})
+		);
+	}
+
+	return Promise.all(updates);
```

Comment:
Do we really need the `Promise.all()` here over awaiting the promise directly in the inside the loop? Since `Promise.all()` will exit on the first promise rejection, which may be halfway the update if you have a large data-model, do we want/need to run this in a transaction? 🤔 Or perhaps use `Promise.allSettled()` instead to ignore failing updates

Reply:
> I don't see a difference here between awaiting in loop and using Promise.all(), in both cases the update might fail at any point.

Indeed they both behave the same in that respect, the point was more that if we directly await the promise you can skip the whole array building part and it would behave the same.

The failing part was more intended to apply to the transaction part.

In any case none of this is blocking.

Reply:
> Indeed they both behave the same in that respect, the point was more that if we directly await the promise you can skip the whole array building part and it would behave the same.

Yeah, they behave almost the same in terms of error, but the `Promise.all()` way leverages full async.

> The failing part was more intended to apply to the transaction part, which they apparently are already by default 😄

I'm afraid migrations aren't executed in transactions, as we aren't relying on `knex.migrate` (was just a ref), so I fully agree with you that this is something we should look into but out of scope of this PR in my opinion.

Reply:
Considering that they arent run in a transaction would it be wise to us `allSettled` instead so that the migration will try to update all possible csv fields even if on fail for unexplainable reasons along the way? 🤔

Reply:
Works for me, feel free to update the PR 👍 
Keep in mind that `allSettled` never rejects, so you need to check each promise result and throw yourself.